hw/xbox/nv2a/nv2a_vsh.c

```diff
@@ -674,13 +674,24 @@ static const char* vsh_header =
     "#define EXP(dest, mask, src) dest.mask = _EXP(_in(src).x).mask\n"
     "vec4 _EXP(float src)\n"
     "{\n"
-    "  return vec4(exp2(src));\n"
+    "  vec4 result;\n"
+    "  result.x = exp2(floor(src));\n"
+    "  result.y = src - floor(src);\n"
+    "  result.z = exp2(src);\n"
```

Comment:
Is this correct? Upon a cursory glance, `result.z` doesn't seem to match what is in the [spec [2.14.1.10.15]](https://www.khronos.org/registry/OpenGL/extensions/NV/NV_vertex_program.txt). It looks like it should be something like:

```c
result.z = result.x * APPX(result.y);

"APPX is an implementation dependent approximation of exponential base 2 such that | exp(q.y*log(2.0))-APPX(q.y) | < 1/(2^11)"
```